src/main.zig

```diff
@@ -1699,7 +1709,9 @@ fn buildOutputType(
                                     continue;
                                 }
                             }
-                            if (mem.eql(u8, linker_arg, "--as-needed")) {
+                            if (mem.eql(u8, linker_arg, "--build-id")) {
+                                build_id = .fast;
```

Comment:
Would it be better to drop this and only allow explicit build-id selections?

Reply:
`ld` allows both. From `ld(1)`:

```
       --build-id
       --build-id=style
```

Reply:
> `ld` allows both. From `ld(1)`:

ld allows it, but we don't need to inherit this behavior.

Since this PR is already backwards incompatible (since it deprecates `-fbuild-id`), maybe its best to drop `--build-id` without a style, so the intended result behind the use of the flag is more intentional.